### PR TITLE
add store id in metric

### PR DESCRIPTION
**Pull request by overvenus**
#629

### CODE REVIEW

**Comment by siddontang** [comment]
Cool, LGTM

PTAL @ngaut

**Comment by BusyJay** [comment]
Cache the generated string maybe better.

File: src/raftstore/store/peer.rs [hunk 1]
```diff
@@ -326,7 +326,7 @@ impl Peer {
         }
 
         debug!("propose command with uuid {:?}", cmd.uuid);
-        metric_incr!("raftstore.propose");
+        metric_incr!(&format!("raftstore.{}.propose", self.peer.get_store_id()));
```
Comment (ngaut) [inline src/raftstore/store/peer.rs]: The method format is kind of slow. Could you maintain a store prefix: raftstore.{store id}

**Comment by overvenus** [comment]
@BusyJay @ngaut 
Do you mean that add a new filed `prefix` in `Peer`?

**Comment by overvenus** [comment]
PTAL @BusyJay @ngaut @siddontang

File: src/raftstore/store/peer.rs [hunk 2]
```diff
@@ -743,6 +746,15 @@ impl Peer {
     }
 }
 
+impl Peer {
+    fn format_metric(&self, s: &str) -> String {
+        let mut m = self.metric_prefix.clone();
+        m.push('.');
```
Comment (siddontang) [inline src/raftstore/store/peer.rs]: I guess this may cause multi re-allocation? @BusyJay

**Comment by siddontang** [comment]
PTAL @BusyJay @disksing

**Comment by disksing** [comment]
LGTM

File: src/raftstore/store/peer.rs [hunk 3]
```diff
@@ -743,6 +746,16 @@ impl Peer {
     }
 }
 
+impl Peer {
```
Comment (siddontang) [inline src/raftstore/store/peer.rs]: why starts a new Peer scope?

**Comment by siddontang** [comment]
PTAL @BusyJay

**Comment by siddontang** [comment]
Hi @overvenus 

I think for a tikv server, all metrics in the server using `tikv.id` as the prefix is ok and more simple, you don't need to care prefix every time now, and only set once in metric client initialization. 

The id is the store id which can identify the tikv uniquely, the store id can be retrieved after node starts (https://github.com/pingcap/tikv/blob/master/src/bin/tikv-server.rs#L274) using `node.id()`, and then you can set metric client here.

**Comment by overvenus** [comment]
@siddontang OK, I will send a PR later.

File: src/raftstore/store/peer.rs (hunk 3 above)
Comment (overvenus) [inline src/raftstore/store/peer.rs]: Tihs method neither belongs to  _commands scope_(below) nor _raft scope_(above)

**Comment by siddontang** [comment]
You can close this PR now.

**Pull request closed without merging**